### DISCUSSION TITLE
Keras 3: Access gradients during training

### DISCUSSION THREAD

**Discussion by santi-mir** [keras-team/keras#20066] [Q&A] [answered]
* Can Keras print / access the calculated gradients during the built-in training loop ? 
* Do custom callbacks for the `fit` function have access to this ?

**Comment by santi-mir** [comment]
I dont think we can access the "updates" but at least the weights we can access:`self.model.get_weights()`

**Comment by fchollet** [comment, marked as answer]
Do get access to gradients, you have to write a custom `train_step()` for your model (you can still use `fit()`). Like this: 

https://keras.io/guides/custom_train_step_in_jax/
https://keras.io/guides/custom_train_step_in_tensorflow/
https://keras.io/guides/custom_train_step_in_torch/

Custom callbacks don't have access to gradients by default.

An alternative can also be to write a custom optimizer, since optimizers have access to gradients.

**Reply by santi-mir** [reply]
Thank you @fchollet ! I love the library ❤️
